modules/illumina.nf

```diff
@@ -102,10 +121,16 @@ process filterResidualAdapters {
     tuple val(sampleName), path(forward), path(reverse)
 
     output:
-    tuple val(sampleName), path("*1_posttrim_filter.fq.gz"), path("*2_posttrim_filter.fq.gz")
+    tuple val(sampleName), path("*1_posttrim_filter.fq.gz"), path("*2_posttrim_filter.fq.gz"), emit: filter_out
+    tuple val(sampleName), path("${sampleName}_filterResidualAdapters_provenance.yml"), emit: provenance
 
     script:
     """
+    printf -- "- process_name: filterResidualAdapters\\n"                                                 >> ${sampleName}_filterResidualAdapters_provenance.yml
+    printf -- "  tools:\\n"                                                                               >> ${sampleName}_filterResidualAdapters_provenance.yml
+    printf -- "    - tool_name: filter_residual_adapters.py\\n"                                           >> ${sampleName}_filterResidualAdapters_provenance.yml
```

Comment:
This is interesting. In general we haven't been collecting provenance for custom scripts. I'm not opposed to it but we might think a bit about how we want to format. I like your approach of using a sha256 of the script as a version.

Reply:
stolen from: @cabrerad :) https://github.com/BCCDC-PHL/AMR-TB-amplicon-artic-nf/blob/main/modules/illumina.nf#L133

Reply:
Ok, in the interest of getting this merged I think we can go with this. But I wonder if we might want to distinguish custom scripts that are packaged in the pipeline from 'tools' like bwa, samtools, etc.